pkg/apis/authorization/types.go

```diff
@@ -17,13 +17,19 @@ limitations under the License.
 package authorization
 
 import (
+	"k8s.io/kubernetes/pkg/api"
 	"k8s.io/kubernetes/pkg/api/unversioned"
 )
 
+// +genclient=true
+// +nonNamespaced=true
+// +noMethods=true
+
 // SubjectAccessReview checks whether or not a user or group can perform an action.  Not filling in a
 // spec.namespace means "in all namespaces".
 type SubjectAccessReview struct {
 	unversioned.TypeMeta
+	api.ObjectMeta
```

Comment:
> What is kubectl create doing that it requires there to be ObjectMeta?
> What would be the minimal changes to kubectl to be able to handle these objects without ObjectMeta?
> 
> I feel like these are not "normal" objects because they are not persisted, and IIUC, cannot be GET-ed. So, maybe the need a different kind of metadata.

The metadata accessor libraries require everything.  In addition, if we use this for delegated authorization, it will be a hot-path and including this type allows us to take advantage of conforming to the `meta.Accessor` fast path.